src/main/java/core/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package core.dao;
+
+import core.db.Storage;
+import core.lib.Dao;
+import core.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return getAll().stream()
+                .filter(m -> m.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer updateManufacturer = getAll().stream()
+                .filter(m -> m.getId().equals(manufacturer.getId()))
+                .findFirst()
+                .get();
+        updateManufacturer.setCountry(manufacturer.getCountry());
+        updateManufacturer.setName(manufacturer.getName());
```

Comment:
this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object

Reply:
done